util/types/json/compare.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"bytes"
+	"reflect"
+	"strings"
+)
+
+// floatEpsilon is for compare two float value allowing precision loss.
+const floatEpsilon float64 = 1.e-8
+
+// compareFloat64 returns an integer comparing the float64 x to y,
+// allowing precision loss.
+func compareFloat64PrecisionLoss(x, y float64) int {
+	if x-y < floatEpsilon && y-x < floatEpsilon {
+		return 0
+	} else if x-y < 0 {
+		return -1
+	}
+	return 1
+}
+
+// jsonTypePrecedences is for comparing two json.
+// See: https://dev.mysql.com/doc/refman/5.7/en/json.html#json-comparison
+var jsonTypePrecedences = map[string]int{
+	"BLOB":     -1,
```

Comment:
we will never convert them to uint. we just want a number list, order by desc. so I begin with -1, and so on -2, -3, ...